spec/models/manageiq/providers/redhat/infra_provider_spec.rb

```diff
@@ -9,13 +12,44 @@
     described_class.description.should == 'Red Hat Enterprise Virtualization Manager'
   end
 
-  it "rhevm_metrics_connect_options" do
-    h = FactoryGirl.create(:ems_redhat, :hostname => "h")
-    h.rhevm_metrics_connect_options[:host].should == "h"
+  context '#rhevm_metrics_connect_options' do
+    it "rhevm_metrics_connect_options" do
+      h = FactoryGirl.create(:ems_redhat, :hostname => "h")
+      h.rhevm_metrics_connect_options[:host].should == "h"
+    end
+
+    it "rhevm_metrics_connect_options overrides" do
+      h = FactoryGirl.create(:ems_redhat, :hostname => "h")
+      h.rhevm_metrics_connect_options(:hostname => "i")[:host].should == "i"
+    end
   end
 
-  it "rhevm_metrics_connect_options overrides" do
-    h = FactoryGirl.create(:ems_redhat, :hostname => "h")
-    h.rhevm_metrics_connect_options(:hostname => "i")[:host].should == "i"
+  context '#connect' do
+    before do
+      class Foo
+        def initialize(_)
+        end
+      end
```

Comment:
I don't get the need for a new class.  Even so, we have to clean up after ourselves if we are creating a new class.  Down below, just return a `:handle` or a double object or something...you don't literally need a class, I think